src/game/logic/object/weapon.cpp

```diff
@@ -101,10 +102,253 @@ void WeaponStore::Parse_Weapon_Template(INI *ini, void *, void *store, const voi
 {
     const char *name = ini->Get_Next_Token();
     const WeaponTemplate *w = g_theWeaponStore->Find_Weapon_Template(name);
-    captainslog_dbgassert(w != nullptr && !strcasecmp(name, "None"), "WeaponTemplate %s not found!", name);
+    captainslog_dbgassert(w != nullptr || !strcasecmp(name, "None"), "WeaponTemplate %s not found!", name);
     *static_cast<const WeaponTemplate **>(store) = w;
 }
 
+Weapon *WeaponStore::Allocate_New_Weapon(const WeaponTemplate *tmpl, WeaponSlotType wslot)
+{
+    return new Weapon(tmpl, wslot);
+}
+
+Weapon::Weapon(const WeaponTemplate *tmpl, WeaponSlotType wslot) :
+    m_template(tmpl),
+    m_wslot(wslot),
+    m_status(OUT_OF_AMMO),
+    m_ammoInClip(0),
+    m_whenWeCanFireAgain(0),
+    m_whenPreAttackFinished(0),
+    m_whenLastReloadStarted(0),
+    m_projectileStreamID(OBJECT_UNK),
+    m_leechWeaponRangeActive(false),
+    m_maxShotCount(0x7FFFFFFF),
+    m_curBarrel(0),
+    m_lastFireFrame(0)
+{
+    m_pitchLimited = m_template->Get_Min_Target_Pitch() > -GAMEMATH_PI || m_template->Get_Max_Target_Pitch() < GAMEMATH_PI;
+    m_numShotsForCurBarrel = m_template->Get_Shots_Per_Barrel();
+    m_suspendFXDelay = m_template->Get_Suspend_FX_Delay() + g_theGameLogic->Get_Frame();
+}
+
+Weapon::Weapon(const Weapon &that) :
+    m_template(that.m_template),
+    m_wslot(that.m_wslot),
+    m_status(OUT_OF_AMMO),
+    m_ammoInClip(0),
+    m_whenWeCanFireAgain(0),
+    m_whenPreAttackFinished(0),
+    m_whenLastReloadStarted(0),
+    m_projectileStreamID(OBJECT_UNK),
+    m_leechWeaponRangeActive(false),
+    m_maxShotCount(0x7FFFFFFF),
+    m_curBarrel(0),
+    m_lastFireFrame(0)
```

Comment:
m_lastFireFrame initializer is in wrong order

Reply:
fixed.